databuilder/extractor/atlas_search_data_extractor.py

```diff
@@ -0,0 +1,279 @@
+             lambda x: AtlasSearchDataExtractor._filter_none(
+                 [c.get('attributes').get('description') for c in x if c.get('status').lower() == 'active']), []),
+            ('tags', 'tags', None, []),
+            ('badges', 'classifications',
+             lambda x: AtlasSearchDataExtractor._filter_none(
+                 [c.get('typeName') for c in x if c.get('entityStatus', '').lower() == 'active']), []),
+            ('display_name', 'attributes.qualifiedName', lambda x: x.split('@')[0], None),
+            ('schema_description', 'attributes.parameters.sourceDescription', None, None),
+            (
+                'programmatic_descriptions', 'attributes.parameters', lambda x: [str(s) for s in list(x.values())],
+                {})
+        ]
+    }
+
+    ENTITY_MODEL_BY_TYPE = {
+        'Table': 'databuilder.models.table_elasticsearch_document.TableESDocument'
+    }
+
+    REQUIRED_RELATIONSHIPS_BY_TYPE = {
+        'Table': ['columns']
+    }
+
+    def init(self, conf: ConfigTree) -> None:
+        self.conf = conf.with_fallback(AtlasSearchDataExtractor.DEFAULT_CONFIG)
+        self.driver = self._get_driver()
+
+        self._extract_iter: Optional[Iterator[Any]] = None
+
+    @property
+    def entity_type(self) -> str:
+        return self.conf.get(AtlasSearchDataExtractor.ENTITY_TYPE_KEY)
+
+    @property
+    def basic_search_query(self) -> Dict:
+        query = {
+            'typeName': self.entity_type,
+            'excludeDeletedEntities': True,
+            'query': '*'
+        }
+
+        LOGGER.debug(f'Basic Search Query: {query}')
+
+        return query
+
+    @property
+    def dsl_search_query(self) -> Dict:
+        query = {
+            'query': f'{self.entity_type} where __state = "ACTIVE"'
+        }
+
+        LOGGER.debug(f'DSL Search Query: {query}')
+
+        return query
+
+    @property
+    def model_class(self) -> Any:
+        model_class = AtlasSearchDataExtractor.ENTITY_MODEL_BY_TYPE.get(self.entity_type)
+
+        if model_class:
+            module_name, class_name = model_class.rsplit(".", 1)
+            mod = importlib.import_module(module_name)
+
+            return getattr(mod, class_name)
+
+    @property
+    def field_mappings(self) -> type_fields_mapping:
+        return AtlasSearchDataExtractor.FIELDS_MAPPING_SPEC.get(self.entity_type) or []
+
+    def extract(self) -> Any:
+        if not self._extract_iter:
+            self._extract_iter = self._get_extract_iter()
+
+        try:
+            return next(self._extract_iter)
+        except StopIteration:
+            return None
+
+    def get_scope(self) -> str:
+        return 'extractor.atlas_search_data'
+
+    @staticmethod
+    def _filter_none(input_list: List) -> List:
+        return list(filter(None, input_list))
+
+    def _get_driver(self) -> Any:
+        return Atlas(host=self.conf.get_string(AtlasSearchDataExtractor.ATLAS_URL_CONFIG_KEY),
```

Comment:
same with this. `self.*`

Reply:
I wouldn't change it since it refers to `class` property not class instance property and this approach is used consistently throughout all objects (extractors/publishers)

Reply:
But you are calling this for this instance only. and not under a static method. 🤷‍♂️

Reply:
Not exactly, please refer to sample_atlas_search_extractor.py to see how those are (or could be) used. Any of those `_KEY` properties can be used as part of config process. it's a general convention used in databuilder afaik and I wouldn't want to make our implementations different.